src/redis_index.c

```diff
@@ -199,8 +201,8 @@ IndexReader *Redis_OpenReader(const RedisSearchCtx *ctx, RSQueryTerm *term, DocT
     goto err;
   }
 
-  if (!idx->numDocs ||
-     (Index_StoreFieldMask(ctx->spec) && !(idx->fieldMask & fieldMask))) {
+  if (!InvertedIndex_NumDocs(idx) ||
+     (Index_StoreFieldMask(ctx->spec) && !(InvertedIndex_FieldMask(idx) & fieldMask))) {
     // empty index! or index does not have results from requested field.
```

Comment:
```suggestion
  if (!InvertedIndex_NumDocs(idx) ||
     !(InvertedIndex_FieldMask(idx) & fieldMask)) {
    // empty index! or index does not have results from requested field.
```
can it be simpliefied since where checking in the getter function?

Reply:
Here we are in `TEXT` context, so for flags under `INDEX_STORAGE_MASK`, it doesn't really matter and we can avoid testing the spec